react-graph/src/Plot.tsx

```diff
@@ -103,13 +111,52 @@ const Plot: React.FunctionComponent<IProps> = (props) => {
     const [heightXLabel, setHeightXLabel] = React.useState<number>(0);
     const [heightYLabel, setHeightYLabel] = React.useState<number>(0);
 
+    // States for Props to avoid change notification on ref change
+    const [defaultTdomain, setDefaultTdomain]= React.useState<[number, number]>(props.defaultTdomain);
+    const [defaultYdomain, setDefaultYdomain] = React.useState<[number,number]| undefined>(props.defaultYdomain);
+    const [updateFlag, setUpdateFlag] = React.useState<number>(0);
     // Constants
     const SVGHeight = props.height - (props.legend === 'bottom'? (props.legendHeight !== undefined? props.legendHeight : 50) : 0);
     const SVGWidth = props.width - (props.legend === 'right'? (props.legendWidth !== undefined? props.legendWidth : 100) : 0);
 
+    const zoomMode = props.zoomMode === undefined? 'AutoValue' : props.zoomMode;
+    
+    // enforce T limits
+    React.useEffect(() => {
+      if (props.Tmin !== undefined && tDomain[0] < props.Tmin)
+        setTdomain((t) => ([props.Tmin ?? 0, t[1]]));
+      if (props.Tmax !== undefined && tDomain[1] > props.Tmax)
+        setTdomain((t) => ([t[0], props.Tmax ?? 0]));
```

Comment:
```suggestion
    React.useEffect(() => {
      if (props.Tmin !== undefined && tDomain[0] < props.Tmin)
        setTdomain((t) => ([props.Tmin, t[1]]));
      if (props.Tmax !== undefined && tDomain[1] > props.Tmax)
        setTdomain((t) => ([t[0], props.Tmax]));
```
nullish coalescing operator seems to be unnecessary since there is the undefined check.
Is this simply to stop typescript/lint from complaining about a type mismatch?